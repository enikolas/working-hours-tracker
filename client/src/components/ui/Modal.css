.modal-layer {
  display: none;
}

.modal-layer.active {
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 100;
  background: rgba(0,0,0,0.3);
}

.modal {
  background: #fff;
  padding: 20px 30px;
  color: #777;
  border-radius: 5px;
  box-shadow: 2px 2px 10px rgba(0,0,0,0.3);
  min-width: 300px;
  position: relative;
}

.modal .close {
  position: absolute;
  right: 10px;
  top: 14px;
  color: #909fa8;
  font-size: 30px;
  line-height: 10px;
  font-weight: bold;
}

.modal .title {
  font-size: 20px;
  padding-bottom: 10px;
  font-weight: bold;
  color: #254356;
}

.modal .content {
  font-size: 16px;
}

.modal .buttons {
  padding: 10px 0 0;
  display: flex;
  align-items: right;
  justify-content: flex-end;
}

.modal .buttons button {
  padding: 8px 12px;
  border-radius: 5px;
  background-color: #254356;
  color: #fff;
  font-size: 16px;
  margin-left: 12px;
}

.modal .buttons button[disabled] {
  opacity: 0.5;
}

.modal .modal-field {
  margin: 5px 0;
}

.modal .modal-field label {
  font-size: 12px;
}

.modal .modal-field input,
.modal .modal-field select {
  display: block;
  background: #c5ccd1;
  padding: 4px 6px;
  margin: 0 auto;
}
