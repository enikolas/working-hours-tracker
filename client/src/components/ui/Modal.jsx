import React from 'react';
import PropTypes from 'prop-types';

import strings from '../../utils/strings';

import './Modal.css';

const showButtons = buttons => (
	<div className="buttons">
		{buttons.map(button => (
			<button
				key={button.label}
				type="button"
				onClick={button.action}
			>
				{button.label}
			</button>
		))}
	</div>
);

const handleLayerClick = cancel => (event) => {
	event.stopPropagation();
	if (cancel && event.target.className.includes('modal-layer')) {
		cancel();
	}
};

/* ESLint directives below commented out to allow for the Modal layer to be clickable */
/* eslint-disable jsx-a11y/no-static-element-interactions,
	jsx-a11y/click-events-have-key-events */

const Modal = props => (
	<div
		className={`modal-layer ${props.active ? 'active' : ''}`}
		onClick={handleLayerClick(props.cancel)}
	>
		<div className="modal">
			{props.cancel ?
				<button className="close" onClick={props.cancel}>&times;</button> : ''
			}
			<div className="title">
				{props.title}
			</div>
			<div className="content">
				{props.content}
			</div>
			{props.hasButtons ?
				showButtons(props.buttons) : ''
			}
		</div>
	</div>
);

export default Modal;

Modal.propTypes = {
	active: PropTypes.bool,
	title: PropTypes.string,
	cancel: PropTypes.func,
	content: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.element
	]),
	hasButtons: PropTypes.bool,
	buttons: PropTypes.arrayOf(PropTypes.shape({
		action: PropTypes.func,
		label: PropTypes.string
	}))
};

Modal.defaultProps = {
	active: false,
	cancel: null,
	title: '',
	content: '',
	hasButtons: true,
	buttons: [{ action: () => {}, label: strings.ok }]
};
