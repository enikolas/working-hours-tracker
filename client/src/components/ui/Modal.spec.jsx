import React from 'react';
import { shallow } from 'enzyme';

import Modal from './Modal';

describe('Modal', () => {
	it('should render properly without any props', () => {
		const wrapper = shallow(<Modal />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should show modal if props active', () => {
		const wrapper = shallow(<Modal active />);
		expect(wrapper).toMatchSnapshot();
	});
	it('shoudnt render buttons if hasbutton false', () => {
		const wrapper = shallow(<Modal hasButtons={false} />);
		expect(wrapper).toMatchSnapshot();
	});
});
