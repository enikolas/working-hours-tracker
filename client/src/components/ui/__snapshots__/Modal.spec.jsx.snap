// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal shoudnt render buttons if hasbutton false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    active={false}
    buttons={
      Array [
        Object {
          "action": [Function],
          "label": "OK",
        },
      ]
    }
    cancel={null}
    content=""
    hasButtons={false}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal"
      >
        
        <div
          className="title"
        >
          
        </div>
        <div
          className="content"
        >
          
        </div>
        
      </div>,
      "className": "modal-layer ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <div
            className="title"
          >
            
          </div>,
          <div
            className="content"
          >
            
          </div>,
          "",
        ],
        "className": "modal",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "title",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "content",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        "",
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal"
        >
          
          <div
            className="title"
          >
            
          </div>
          <div
            className="content"
          >
            
          </div>
          
        </div>,
        "className": "modal-layer ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <div
              className="title"
            >
              
            </div>,
            <div
              className="content"
            >
              
            </div>,
            "",
          ],
          "className": "modal",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "title",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "content",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          "",
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Modal should render properly without any props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    active={false}
    buttons={
      Array [
        Object {
          "action": [Function],
          "label": "OK",
        },
      ]
    }
    cancel={null}
    content=""
    hasButtons={true}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal"
      >
        
        <div
          className="title"
        >
          
        </div>
        <div
          className="content"
        >
          
        </div>
        <div
          className="buttons"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            OK
          </button>
        </div>
      </div>,
      "className": "modal-layer ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <div
            className="title"
          >
            
          </div>,
          <div
            className="content"
          >
            
          </div>,
          <div
            className="buttons"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              OK
            </button>
          </div>,
        ],
        "className": "modal",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "title",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "content",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
                type="button"
              >
                OK
              </button>,
            ],
            "className": "buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "OK",
              "nodeType": "host",
              "props": Object {
                "children": "OK",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "OK",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal"
        >
          
          <div
            className="title"
          >
            
          </div>
          <div
            className="content"
          >
            
          </div>
          <div
            className="buttons"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              OK
            </button>
          </div>
        </div>,
        "className": "modal-layer ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <div
              className="title"
            >
              
            </div>,
            <div
              className="content"
            >
              
            </div>,
            <div
              className="buttons"
            >
              <button
                onClick={[Function]}
                type="button"
              >
                OK
              </button>
            </div>,
          ],
          "className": "modal",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "title",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "content",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  onClick={[Function]}
                  type="button"
                >
                  OK
                </button>,
              ],
              "className": "buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "OK",
                "nodeType": "host",
                "props": Object {
                  "children": "OK",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "OK",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Modal should show modal if props active 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    active={true}
    buttons={
      Array [
        Object {
          "action": [Function],
          "label": "OK",
        },
      ]
    }
    cancel={null}
    content=""
    hasButtons={true}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal"
      >
        
        <div
          className="title"
        >
          
        </div>
        <div
          className="content"
        >
          
        </div>
        <div
          className="buttons"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            OK
          </button>
        </div>
      </div>,
      "className": "modal-layer active",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <div
            className="title"
          >
            
          </div>,
          <div
            className="content"
          >
            
          </div>,
          <div
            className="buttons"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              OK
            </button>
          </div>,
        ],
        "className": "modal",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "title",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "content",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
                type="button"
              >
                OK
              </button>,
            ],
            "className": "buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "OK",
              "nodeType": "host",
              "props": Object {
                "children": "OK",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "OK",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal"
        >
          
          <div
            className="title"
          >
            
          </div>
          <div
            className="content"
          >
            
          </div>
          <div
            className="buttons"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              OK
            </button>
          </div>
        </div>,
        "className": "modal-layer active",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <div
              className="title"
            >
              
            </div>,
            <div
              className="content"
            >
              
            </div>,
            <div
              className="buttons"
            >
              <button
                onClick={[Function]}
                type="button"
              >
                OK
              </button>
            </div>,
          ],
          "className": "modal",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "title",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "content",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  onClick={[Function]}
                  type="button"
                >
                  OK
                </button>,
              ],
              "className": "buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "OK",
                "nodeType": "host",
                "props": Object {
                  "children": "OK",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "OK",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
